Backport fixes for cms

smuzaffar opened the pull request.

Axel-Naumann committed:
[root-config] Fix --git-revision:

Fixes https://github.com/root-project/root/issues/14229

Axel-Naumann committed:
[cmake] gitinfo.txt: also provide commit line for tags.

vgvassilev committed:
Revert "[core] Materialize symbols for TError variables"

This reverts commit 329fb5ae0c5717aec95fe50ec276f07450ac7f11.

vgvassilev committed:
[core] Prefer variable initializer over symbol search

Symbol lookup is a quite expensive operation and might result in JIT
compilation and library loading.

Co-authored-by: Jonas Hahnfeld <jonas.hahnfeld@cern.ch>

vgvassilev committed:
[core] Reduce symbol search only to when autoloading is enabled.

The llvm9 JIT issued callbacks when a symbol was missing and we reacted on it
by loading the relevant library. In root-project/root@9b2041e3 we have kept the
logic but now the JIT started querying more often even for symbols which are
okay to be missing. In turn that leads to scanning all libraries causing
performance issues.

This patch tries to limit this functionality only in contexts where automatic
loading is allowed.

vgvassilev committed:
[core] Evaluate initializer of constant variables

Co-authored-by: Jonas Hahnfeld <jonas.hahnfeld@cern.ch>

vepadulano committed:
[core] Make some constants constexpr

This allows us to avoid generating symbols in libCore for these constants
keeping the same amount of open calls at ROOT startup time.

smuzaffar commented:
please test

cmsbuild commented:
A new Pull Request was created by  @smuzaffar (Malik Shahzad Muzaffar) for branch cms/v6-30-00-patches/0541c6f.

@iarspider, @aandvalenzuela, @smuzaffar can you please review it and eventually sign? Thanks.
@rappoccio, @antoniovilela, @sextonkennedy you are the release manager for this.
cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"14cc85ca31da6a191331866ada9e132500d6bc79":{"files":["core/foundation/inc/TError.h","core/foundation/src/RLogger.cxx","core/foundation/src/TError.cxx"],"squashed":false,"time":1705434546},"284c5d5e5e1db8b657b5f493a5c9da4b1f46eef0":{"files":["core/base/inc/TString.h","core/base/src/TString.cxx","core/foundation/inc/RtypesCore.h"],"squashed":false,"time":1705434546},"7312bdacfb2e10fb34d814bac016fb776009b125":{"files":["cmake/modules/SetROOTVersion.cmake"],"squashed":false,"time":1704811791},"a2be28ed4acf4587580aeaab299030d6478ee789":{"files":["core/metacling/src/TClingDataMemberInfo.cxx"],"squashed":false,"time":1705434546},"a6ca1c3b75c9ef42d65b713036a92a849e967ac2":{"files":["core/metacling/src/TCling.cxx","core/metacling/src/TClingCallbacks.cxx","core/metacling/src/TClingCallbacks.h","core/metacling/test/TClingDataMemberInfoTests.cxx"],"squashed":false,"time":1705434546},"ce9bbc826d26b8a5275221e31e2eef5e6c588e7a":{"files":["core/metacling/src/TClingDataMemberInfo.cxx"],"squashed":false,"time":1705434546},"d2a9b574118db9f12acae8a8f2dd88ce1b2169ca":{"files":["config/root-config.in"],"squashed":false,"time":1704811791},"d4f8f6aa802bacb0dd592ae66c90f2a1eeb0cb07":{"files":["core/metacling/test/TClingDataMemberInfoTests.cxx"],"squashed":false,"time":1705434584}},"emoji":{},"last_seen_sha":"d4f8f6aa802bacb0dd592ae66c90f2a1eeb0cb07","signatures":{}} -->

cmsbuild commented:
Pull request #195 was updated.

smuzaffar commented:
please test

vepadulano committed:
[core] Fix statements at global scope in TCling test

`TInterpreter::Declare` does not support issuing statements on the global scope, while `TInterpreter::ProcessLine` does. This test was first introduced in a development version of ROOT (6.31), where the upgrade to LLVM16 was already in place. In that scenario, clang supports global scope statements (thanks to changes made for clang-repl). Applying the test to ROOT 6.30 uncovered the problem, since the clang of LLVM13 does not support global scope statements. Fix the test by using `ProcessLine` instead of `Declare` which does the intended thing independently from the ROOT version.

Co-authored-by: Vassil Vassilev <vasil.georgiev.vasilev@cern.ch>

cmsbuild commented:
Pull request #195 was updated.

smuzaffar commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-860910/36876/summary.html
**COMMIT**: d4f8f6aa802bacb0dd592ae66c90f2a1eeb0cb07
**CMSSW**: CMSSW_14_0_X_2024-01-16-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/195/36876/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test testAlcaBeamMonitor had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [added 87 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-16-2300+860910/60610/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 6 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-16-2300+860910/60610/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3247526
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-16-2300+860910/60610/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3247501
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 200 log files, 161 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-16-2300+860910/60610/triggerResults): no differences found
